### PR TITLE
Data Plane Update 2022-01-01.15.0

### PR DESCRIPTION
**Pull request by seth-barshay**
# Description

[Swagger Pull Request](https://github.com/Azure/azure-rest-api-specs/pull/16960)

- Added property uploadHeaders to OutputFileBlobContainerDestination.
  - Allows users to set custom HTTP headers on resource file uploads.
  - Array of type HttpHeader (also being added).
- Added boolean property allowTaskPreemption to JobSpecification, CloudJob, JobAddParameter, JobPatchParameter, JobUpdateParameter
  - Mark Tasks as preemptible for higher priority Tasks (requires Comms-Enabled or Single Tenant Pool).
- Replaced comment (title, description, etc.) references of "low-priority" with "Spot/Low-Priority", to reflect new service behavior.
  - No API change required.
  - Low-Priority Compute Nodes (VMs) will continue to be used for User Subscription pools (and only User Subscription pools), as before.
  - Spot Compute Nodes (VMs) will now be used for Batch Managed (and only Batch Managed pools) pools.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x]  **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### CODE REVIEW

File: sdk/batch/microsoft-azure-batch/src/test/java/com/microsoft/azure/batch/FileTests.java [hunk 1]
```diff
@@ -87,10 +87,10 @@ public String call(InputStream input) {
                 Assert.assertEquals("hello\n", output);
 
                 //Running this check temporarily in Record mode only, playback mode parses incorrect value from the recording.
-                if(isRecordMode()) {
-                    FileProperties properties = batchClient.fileOperations().getFilePropertiesFromTask(jobId, taskId, "stdout.txt");
-                    Assert.assertEquals(6, properties.contentLength());
-                }
+                //if(isRecordMode()) {
```
Comment (dpwatrous) [inline sdk/batch/microsoft-azure-batch/src/test/java/com/microsoft/azure/batch/FileTests.java:90]: Why is this (and line 158 below) commented out?

File: sdk/batch/microsoft-azure-batch/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,21 @@
 
 ## 10.1.0-beta.1 (Unreleased)
 
+### Features Added
```
Comment (dpwatrous) [inline sdk/batch/microsoft-azure-batch/CHANGELOG.md:5]: Should we change the version here to 10.1.0 with a date, or is that handled by a separate process?

File: sdk/batch/microsoft-azure-batch/src/test/java/com/microsoft/azure/batch/FileTests.java (hunk 1 above)
Comment (seth-barshay) [reply sdk/batch/microsoft-azure-batch/src/test/java/com/microsoft/azure/batch/FileTests.java:90]: That was accidentally committed--will update now

File: sdk/batch/microsoft-azure-batch/CHANGELOG.md (hunk 2 above)
Comment (seth-barshay) [reply sdk/batch/microsoft-azure-batch/CHANGELOG.md:5]: I think in past PRs the date wasn't changed until it was approved by the SDK team, but I could be wrong about that. Regardless I do think the version should be changed

Comment (seth-barshay) [reply sdk/batch/microsoft-azure-batch/CHANGELOG.md:5]: Just checked and it does seem like that's the case re: the date

**Review by dpwatrous — APPROVED** [review]

**Pull request closed without merging**